src/lj_str.c

```diff
@@ -189,7 +146,12 @@ GCstr *lj_str_new(lua_State *L, const char *str, size_t lenx)
     return &g->strempty;
   }
 
-  h = LJ_STR_HASH(str, lenx);
+#if LJ_OR_STRHASHCRC32
```

Comment:
This should just use g->strhashfn, which should point to lj_str_hash_orig if the underlying CPU does not have the necessary support.  It's an extra indirection, but it shouldn't really matter too much if the hash function is hot.  In fact, you're already doing that in `lj_init_str_hashfn` so you're good to go.

Reply:
This patch is implemented with maintainability and merge-compatibility in mind (and does both successfully imho), hence, the `strhashfn` member does not exist if the binary is built without the optimization, so this approach is not valid. Closely following the upstream implementation allows to distinguish the OpenResyt-added portions of the code, and guarantees backwards-compatibility in case an issue is to be discovered and the binary needs to be rebuilt without the optimization. This approach is much safer and more maintainable.